clouddriver-core/src/main/groovy/com/netflix/spinnaker/clouddriver/cache/CatsOnDemandCacheUpdater.groovy

```diff
@@ -57,13 +61,27 @@ class CatsOnDemandCacheUpdater implements OnDemandCacheUpdater {
   OnDemandCacheUpdater.OnDemandCacheStatus handle(OnDemandAgent.OnDemandType type, Collection<OnDemandAgent> onDemandAgents, Map<String, ? extends Object> data) {
     boolean hasOnDemandResults = false
     for (OnDemandAgent agent : onDemandAgents) {
+      boolean atomicScheduler = false;
+      AgentScheduler scheduler = null
+      if (agent instanceof AgentSchedulerAware) {
+        scheduler = agent.agentScheduler
```

Comment:
I'd be willing to go with the constraint that we have a single scheduler just to ensure the proper semantics when the scheduler is atomic are captured in this one class